tests/HttpServerTest.php

```diff
@@ -428,15 +443,21 @@ public function testBiggerBodyThanContentLengthWillBeCutted()
 
         $socket->emit('connection', array($this->connection));
 
-        $this->connection->expects($this->once())->method('write')->with($this->equalTo("HTTP/1.1 200 OK\r\n\r\n"));
+        $this->connection->expects($this->once())->method('write')->with($this->equalTo("HTTP/1.1 200 OK\r\nContent-Length: 5\r\n\r\nhello"));
 
         $this->connection->emit('data', array($request));
     }
 
     public function testContentLengthInRequestIsZero()
     {
         $callback = function(RequestInterface $request) {
-            return new Response();
+            $promise = new Promise(function ($resolve, $reject) use ($request) {
+                $request->getBody()->on('end', function () use ($resolve) {
```

Comment:
See also above. We likely need to explicitly ensure no `data` event will be emitted here. May also affect other tests afaict?